### DISCUSSION TITLE
How to query manyToMany pivot table with extra condition?

### DISCUSSION THREAD

**Discussion by melokki** [adonisjs/core#1950] [Help]
Hello,
I am stuck in the following problem.
I have multiple many-to-many relationships and I am trying to make a query to get the correct data from database, but I've hit the following wall

My relationships are like this:

Course -> ManyToMany -> Subscription
Subscription -> ManyToMany -> User (in this pivot table I have the courseId also)

Now, I am trying to make a query to get all the courses each one attached with its subscriptions, and each subscription attached with its users

This is what I have so far, but it does not give me the correct result. For instance `course1` will have the same users as `course2` because both courses share the same subscription
 
```ts
const courses = Course.query()
        .preload('subscriptions', (query) => {
          query.preload('users')
        })
```

So I need to add a condition to also check the `course_id` from the `subscription_user` pivot table
I figured I need to add something like this
```ts
query.wherePivot('course_id', 'how do I get the course id?')
```
but I don't know how to get each course id in this where condition or an array with all course ids

Any suggestion?

**Comment by iammrsea** [comment]
The ```wherePivot()``` method would not work in this scenario since there's no way to get the ```course_id```, but what you could do is to also
preload your pivot data, which in this case is the ```courseId```, await the query to complete and transform the courses array returned to the shape you would want based on the ```courseId```. That should be pretty straight forward, a little bit of map, filter or reduce could do the magic.
